tasks/upload.py

```diff
@@ -365,48 +374,26 @@ def run_impl_within_lock(
         *args,
         **kwargs,
     ):
-        log.info(
+        self.log.info(
             "Starting processing of report",
-            extra=dict(
-                repoid=upload_context.repoid,
-                commit=upload_context.commitid,
-                report_type=upload_context.report_type.value,
-                report_code=upload_context.report_code,
-            ),
         )
         if not upload_context.has_pending_jobs():
-            log.info("No pending jobs. Upload task is done.")
+            self.log.info("No pending jobs. Upload task is done.")
             return {
                 "was_setup": False,
                 "was_updated": False,
                 "tasks_were_scheduled": False,
             }
 
-        upload_processing_delay = get_config("setup", "upload_processing_delay")
-        if upload_processing_delay is not None:
-            upload_processing_delay = int(upload_processing_delay)
-            last_upload_timestamp = upload_context.last_upload_timestamp()
-            if last_upload_timestamp is not None:
-                last_upload = datetime.fromtimestamp(
-                    float(last_upload_timestamp), timezone.utc
-                )
-                if (
-                    datetime.now(timezone.utc)
-                    - timedelta(seconds=upload_processing_delay)
-                    < last_upload
-                ):
-                    retry_countdown = max(30, upload_processing_delay)
-                    log.info(
-                        "Retrying due to very recent uploads.",
-                        extra=dict(
-                            repoid=upload_context.repoid,
-                            commit=upload_context.commitid,
-                            report_type=upload_context.report_type.value,
-                            countdown=retry_countdown,
-                        ),
-                    )
-                    upload_context.prepare_kwargs_for_retry(kwargs)
-                    self.retry(countdown=retry_countdown, kwargs=kwargs)
+        if retry_countdown := _should_debounce_processing(upload_context):
```

Comment:
Nice to see this syntax in, last time I tried we were in an older python version an it broke production 🙃. Good reminder that we can use this!